tidy3d/components/simulation.py

```diff
@@ -2716,20 +2716,67 @@ def _projection_monitors_2d(cls, val, values):
                     )
 
+                if isinstance(monitor, FieldProjectionCartesianMonitor):
+                    if plane == "y-z":
+                        if monitor.proj_axis == 0:
+                            raise SetupError(
+                                "For a 2D simulation in the y-z plane, the 'proj_axis' of "
+                                f"monitor '{monitor.name}' should be set to '1' or '2'."
+                            )
+                        elif monitor.proj_axis in (1, 2) and (
+                            len(monitor.x) != 1 or monitor.x[0] != 0
+                        ):
+                            raise SetupError(
+                                "For a 2D simulation in the y-z plane with "
+                                f"'proj_axis = {monitor.proj_axis}', 'x' of monitor "
+                                f"'{monitor.name}' should be set to 0."
+                            )
+                    elif plane == "x-z":
+                        if monitor.proj_axis == 1:
+                            raise SetupError(
+                                "For a 2D simulation in the x-z plane, the 'proj_axis' of "
+                                f"monitor '{monitor.name}' should be set to '0' or '2'."
+                            )
+                        elif monitor.proj_axis == 0 and (len(monitor.x) != 1 or monitor.x[0] != 0):
+                            raise SetupError(
+                                "For a 2D simulation in the x-z plane with "
+                                f"'proj_axis = {monitor.proj_axis}', 'x' of monitor "
+                                f"'{monitor.name}' should be set to 0."
+                            )
+                        elif monitor.proj_axis == 2 and (len(monitor.y) != 1 or monitor.y[0] != 0):
+                            raise SetupError(
+                                "For a 2D simulation in the x-z plane with "
+                                f"'proj_axis = {monitor.proj_axis}', 'y' of monitor "
+                                f"'{monitor.name}' should be set to 0."
+                            )
+                    elif plane == "x-y":
+                        if monitor.proj_axis == 2:
+                            raise SetupError(
+                                "For a 2D simulation in the x-y plane, the 'proj_axis' of "
+                                f"monitor '{monitor.name}' should be set to '0' or '1'."
+                            )
+                        elif monitor.proj_axis in (0, 1) and (
+                            len(monitor.y) != 1 or monitor.y[0] != 0
+                        ):
+                            raise SetupError(
+                                "For a 2D simulation in the x-y plane with "
+                                f"'proj_axis = {monitor.proj_axis}', 'y' of monitor "
+                                f"'{monitor.name}' should be set to 0."
+                            )
```

Comment:
This seems really verbose, maybe something like this would work too?

```python
config = {
    "y-z": {"index": 0, "coord": "x"},
    "x-z": {"index": 1, "coord": "y"},
    "x-y": {"index": 2, "coord": "z"},
}[plane]

invalid_proj_axis = config["index"]
valid_proj_axes = [i for i in range(3) if i != invalid_proj_axis]

if monitor.proj_axis == invalid_proj_axis:
    raise SetupError(
        f"For a 2D simulation in the {plane} plane, the 'proj_axis' of "
        f"monitor '{monitor.name}' should be set to one of {valid_proj_axes}."
    )

coord = getattr(monitor, config["coord"])
if monitor.proj_axis in valid_proj_axes and (len(coord) != 1 or coord[0] != 0):
    raise SetupError(
        f"For a 2D simulation in the {plane} plane with "
        f"'proj_axis = {monitor.proj_axis}', '{config['coord']}' of monitor "
        f"'{monitor.name}' should be set to 0."
    )
```

Reply:
Thanks @yaugenst-flex ! This is very helpful, I will revise it.